File: src/1920.py
import logging
import sys
from bisect import bisect_left, bisect_right
from collections import Counter, defaultdict, deque
from functools import cache, lru_cache
from heapq import heapify, heappop, heappush
from itertools import accumulate
from typing import List, Optional


def logging_setting():
    LOG_LEVEL = logging.INFO    # DEBUG, INFO, WARNING, ERROR, CRITICAL
    if __debug__:
        LOG_LEVEL = logging.DEBUG

    logging.basicConfig(
        stream=sys.stderr,
        level=LOG_LEVEL,
        format="%(levelname)-6s %(asctime)s %(module)s:%(funcName)s:%(lineno)-4s %(message)s",
        datefmt='%Y/%m/%d %H:%M:%S')


class Solution:
    def buildArray(self, nums: List[int]) -> List[int]:
        retVal = [0 for _ in nums]

        numsSize = len(nums)
        for i in range(numsSize):
            retVal[i] = nums[nums[i]]

        return retVal


if __name__ == "__main__":
    logging_setting()

    try:
        logging.info("sys.version: %s", sys.version)
        print()

        pSolution = Solution()
        for nums in [[0, 2, 1, 5, 3, 4], [5, 0, 1, 2, 3, 4]]:
            # /* Example
            #  *  Input: nums = [0,2,1,5,3,4]
            #  *  Output: [0,1,2,4,5,3]
            #  *
            #  *  Input: nums = [5,0,1,2,3,4]
            #  *  Output: [4,5,0,1,2,3]
            #  */
            logging.info("Input: nums = %s", nums)

            retVal = pSolution.buildArray(nums)
            logging.info("Output: %s", retVal)

            print()
    except KeyboardInterrupt as exception:
        logging.error("%s: %s", exception.__class__.__name__, exception, exc_info=True)
        pass
    except Exception as exception:
        logging.error("%s: %s", exception.__class__.__name__, exception, exc_info=True)
        pass

    sys.exit(0)

File: src/1920.cpp
#include <algorithm>
#include <iostream>
#include <vector>

using namespace std;

class Solution {
   public:
    vector<int> buildArray(vector<int>& nums) {
        vector<int> retVal;

        int numsSize = nums.size();
        for (int i = 0; i < numsSize; ++i) {
            retVal.emplace_back(nums[nums[i]]);
        }

        return retVal;
    }
};

int main(int argc, char** argv) {
    struct subject {
        vector<int> nums;
    };
    vector<subject> testData{{{0, 2, 1, 5, 3, 4}}, {{5, 0, 1, 2, 3, 4}}};
    int numberOfTestCase = testData.size();
    /* Example
     *  Input: nums = [0,2,1,5,3,4]
     *  Output: [0,1,2,4,5,3]
     *
     *  Input: nums = [5,0,1,2,3,4]
     *  Output: [4,5,0,1,2,3]
     */

    Solution cSolution;
    vector<int> answer;
    for (int i = 0; i < numberOfTestCase; ++i) {
        cout << "Input: nums = [";
        for (long unsigned int j = 0; j < testData[i].nums.size(); ++j) {
            cout << ((j == 0) ? "" : ",") << testData[i].nums[j];
        }
        cout << "]\n";

        answer = cSolution.buildArray(testData[i].nums);
        cout << "Output: [";
        for (long unsigned int j = 0; j < answer.size(); ++j) {
            cout << ((j == 0) ? "" : ",") << answer[j];
        }
        cout << "]\n";

        cout << "\n";
    }

    return EXIT_SUCCESS;
}
